docs/tools/config_system.md

```diff
@@ -1,23 +1,136 @@
 ## The configuration system
 
-The Arm Mbed configuration system customizes the compile time configuration of various Mbed components (targets, libraries and applications). Each component can define a number of *configuration parameters*. The values of these configuration parameters can then be *overridden* in various ways. Configuration is defined using [JSON](http://www.json.org/). Some examples of configuration parameters:
+The Arm Mbed configuration system, a part of the Arm Mbed Build Tools, customizes compile time configuration parameters. Each component can define a number of configuration parameters in its `mbed_lib.json`. The values of these configuration parameters may be overridden by `mbed_app.json`. Configuration is defined using [JSON](http://www.json.org/). Some examples of configuration parameters:
 
 - The sampling period for a data acquisition application.
 - The default stack size for a newly created OS thread.
 - The receive buffer size of a serial communication library.
 - The flash and RAM memory size of an Mbed target.
 
-The configuration system gathers and interprets all the configurations defined in the source tree. The output of the configuration system is a list of macros that are automatically defined when compiling the code.
+The Mbed configuration system gathers and interprets the configuration defined in the target, all `mbed_lib.json` files and the `mbed_app.json` file. The configuration system creates a single header file, `mbed_config.h`, in the `BUILD` directory that contains all of the defined configuration parameters converted into C preprocessor macros. The Mbed OS configuration system is run during `mbed compile` before invoking the compiler and during `mbed export` before creating project files.
```

Comment:
@loverdeg-ep It's all in Mbed OS's `tools/config/` directory. This is an implementation detail that has changed in the past and I would like to avoid documenting an implementation detail. 

I'll expand on the location of `mbed_config.h`

Reply:
Kind of off topic; Considering this how would I manage the configuration for a project that doesn't have mbedOS as a dependency?

Reply:
@theotherjimmy Was hoping have the question in the previous comment resolved.

Reply:
> Considering this how would I manage the configuration for a project that doesn't have mbedOS as a dependency?

You could use the standalone Mbed 2 tools which contain the same config system. Would you like a mention that the config system is an Mbed OS feature, and may not work without mbed OS in your project?

Reply:
Sure. Anything to allow me to better understand my use cases.

Test framework depends on mbedOS as well?

Reply:
Yep.